sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/FilterEstimation.scala

```diff
@@ -311,6 +311,16 @@ case class FilterEstimation(plan: Filter) extends Logging {
       logDebug("[CBO] No statistics for " + attr)
       return None
     }
+
+    attr.dataType match {
+      case _: NumericType | DateType | TimestampType | BooleanType =>
+        if (!colStatsMap.hasMinMaxStats(attr)) {
+          logDebug(s"[CBO] No statistics for $attr")
+          return None
+        }
+      case StringType | BinaryType =>
```

Comment:
Just `case _ => // do nothing` if you mean to do nothing otherwise? this match would fail on other types not mentioned here otherwise

Reply:
> Just `case _ => // do nothing` if you mean to do nothing otherwise? this match would fail on other types not mentioned here otherwise

Done